Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTraceViewModelTrace.swift

```diff
@@ -114,6 +67,63 @@ extension UtilityNetworkTraceViewModel {
     }
 }
 
+extension UtilityNetworkTraceViewModel.Trace {
+    /// - Parameter name: A name of a utility asset group.
+    /// - Returns: The set of utility elements returned by the trace that belong to the provided
+    /// asset group, grouped by type.
+    func elementsByTypeInGroup(named name: String) -> [String: [UtilityElement]] {
+        let assetsInGroup = elementsInAssetGroup(named: name)
+        var result = [String : [UtilityElement]]()
+        assetsInGroup.forEach { e in
+            var assetTypeGroup = result[e.assetType.name, default: []]
+            assetTypeGroup.append(e)
+            result.updateValue(assetTypeGroup, forKey: e.assetType.name)
+        }
+        return result
```

Comment:
Consider using `reduce(into:_:)`.

```suggestion
        return elementsInAssetGroup(named: name)
            .reduce(into: [:]) { result, element in
                let key = element.assetType.name
                var assetTypeGroup = result[key, default: []]
                assetTypeGroup.append(element)
                result[key] = assetTypeGroup
            }
```